compiler/rustc_mir/src/transform/promote_consts.rs

```diff
@@ -623,6 +629,7 @@ impl<'tcx> Validator<'_, 'tcx> {
 
             Rvalue::NullaryOp(NullOp::Box, _) => return Err(Unpromotable),
 
+            // FIXME(RalfJung): the rest is *implicitly considered promotable*... that seems dangerous.
```

Comment:
yea, we should totally exhaustively match here

Reply:
I'll make that a separate PR with the other refactoring, as there is a big risk of regression there.